application/src/main/java/org/opentripplanner/service/vehicleparking/VehicleParkingRepository.java

```diff
@@ -0,0 +1,20 @@
+package org.opentripplanner.service.vehicleparking;
+
+import com.google.common.collect.ImmutableListMultimap;
+import java.util.Collection;
+import java.util.stream.Stream;
+import org.opentripplanner.service.vehicleparking.model.VehicleParking;
+import org.opentripplanner.service.vehicleparking.model.VehicleParkingGroup;
+
+/**
+ * The writable data store of parking facilities.
+ */
+public interface VehicleParkingRepository {
+  void updateVehicleParking(
+    Collection<VehicleParking> parkingToAdd,
+    Collection<VehicleParking> parkingToRemove
+  );
+  Stream<VehicleParking> getVehicleParkings();
+
+  ImmutableListMultimap<VehicleParkingGroup, VehicleParking> getVehicleParkingGroups();
```

Comment:
Could this return `ListMultimap` - the interface, not the implementing class `ImmutableListMultimap`?

Reply:
It can: https://github.com/opentripplanner/OpenTripPlanner/pull/6246/commits/dd363aab68cacb173a5ada49a4bd888c19af329b